Sync/RustSyncManagerAPI.swift

```diff
@@ -0,0 +1,81 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+import Logger
+import Shared
+@_exported import MozillaAppServices
+
+open class RustSyncManagerAPI {
+    private let logger: Logger
+    let api: SyncManager
+
+    public init(logger: Logger = DefaultLogger.shared) {
+        self.api = SyncManager()
+        self.logger = logger
+    }
+
+    public func disconnect() {
+        DispatchQueue.global(qos: .default).async {
+            self.api.disconnect()
+        }
+    }
+
+    public func sync(
+        params: SyncParams,
+        completion: @escaping (MozillaAppServices.SyncResult) -> Void
+    ) {
```

Comment:
nit: odd indentation fix

```suggestion 
    public func sync(params: SyncParams,
                     completion: @escaping (MozillaAppServices.SyncResult) -> Void) {
```